core/src/main/java/org/apache/calcite/rel/rules/SemiJoinJoinTransposeRule.java

```diff
@@ -102,16 +101,18 @@ public SemiJoinJoinTransposeRule(RelBuilderFactory relBuilderFactory) {
 
     // determine which operands below the semi-join are the actual
     // Rels that participate in the semi-join
+    final ImmutableIntList leftKeys = semiJoin.analyzeCondition().leftKeys;
```

Comment:
It is unnecessary to move this line here?

Reply:
The `leftKeys` variable was defined much earlier than the place where it was actually used. I moved it closer to this place to improve code readability.

Reply:
OK, it make sense